src/Http/Http.Extensions/src/ProblemDetailsServiceCollectionExtensions.cs

```diff
@@ -39,6 +41,15 @@ public static IServiceCollection AddProblemDetails(this IServiceCollection servi
         services.TryAddSingleton<IProblemDetailsService, ProblemDetailsService>();
         services.TryAddEnumerable(ServiceDescriptor.Singleton<IProblemDetailsWriter, DefaultProblemDetailsWriter>());
 
+        services.PostConfigure<JsonOptions>((options) =>
+        {
+            if (options.SerializerOptions.TypeInfoResolver is not null)
```

Comment:
What happens if TypeInfoResolver is null?

Also, what happens if `AddProblemDetails` is called multiple times?